File: dec01/calibration.mjs
// https://adventofcode.com/2018/day/1

import { readFileSync } from 'fs';

const input = readFileSync('./input.txt', 'utf8');
const inputArray = input.split('\n');

let code = 0;

const first_digit = line => line.match(/\d/)[0][0];

inputArray.filter(line => line.trim().length > 0).forEach(
    line => code += parseInt(first_digit(line) + first_digit(Array.from(line).reverse().join('')))
);

console.log(code);

File: dec01/calibration.py
# https://adventofcode.com/2023/day/1

def first_digit(line):
    for c in line:
        if c.isdigit():
            return c
    raise ValueError(f'No digits in: {line}')

code = 0
for i, line in enumerate(open('input.txt'), start=1):
    if not line.strip():
        continue
    digits = first_digit(line) + first_digit(reversed(line))
    code += int(digits)

print(code)
